plugins/linguist/README.md

```diff
@@ -0,0 +1,152 @@
+kind: Component
+metadata:
+  # ...
+  annotations:
+    backstage.io/linguist: https://github.com/backstage/backstage
+spec:
+  type: service
+  # ...
+```
+
+### Frontend
+
+To setup the Linguist Card frontend you'll need to do the following steps:
+
+1. First we need to add the `@backstage/plugin-linguist` package to your frontend app:
+
+   ```sh
+   # From your Backstage root directory
+   yarn add --cwd packages/app @backstage/plugin-linguist
+   ```
+
+2. Second we need to add the `EntityLinguistCard` extension to the entity page in your app:
+
+   ```tsx
+   // In packages/app/src/components/catalog/EntityPage.tsx
+   import { isLinguistAvailable, EntityLinguistCard } from '@backstage/plugin-linguist';
+
+   // For example in the Overview section
+   const overviewContent = (
+    <Grid container spacing={3} alignItems="stretch">
+        // ...
+        <EntitySwitch>
+            <EntitySwitch.Case if={isLinguistAvailable}>
+                <Grid item md={6}>
+                    <EntityLinguistCard />
+                </Grid>
+            </EntitySwitch.Case>
+        </EntitySwitch>
+        // ...
+    </Grid>
+   ```
+
+**Notes:**
+
+- The `if` prop is optional on the `EntitySwitch.Case`, you can remove it if you always want to see the tab even if the entity being viewed does not have the needed annotation
+
+## Permissions
+
+The Linguist plugin supports the [permissions framework](https://backstage.io/docs/permissions/overview), the following sections outline how you can use them with the assumption that you have the permissions framework setup and working.
+
+**Note:** These sections are intended as guidance and are completely optional. The Linguist plugin will work with the permission framework off or on without any specific policy setup.
+
+### Refresh Button
+
+The linguist card features a refresh button that will generate the language breakdown when it has not been generated yet or re-generate the data if the user feels it is not up to date. This is a somewhat expensive operation and you may want to limit who has the ability to trigger this. Here's how you can do that:
+
+1. First we need to add the `@backstage/plugin-linguist-common` package to your backend:
+
+   ```sh
+   # From your Backstage root directory
+   yarn add --cwd packages/backend @backstage/plugin-linguist-common
+   ```
+
+2. Then open `packages/backend/src/plugins/permission.ts`
+3. Now we'll add the following import at the bottom of all the existing import statements:
+
+   ```ts
+   import { linguistUpdateLanguagesPermission } from '@backstage/plugin-linguist-common'
+   ```
+
+4. Next we need to update the `@backstage/plugin-permission-common` import to include `isPermission`:
+
+   ```diff
+     import {
+        AuthorizeResult,
+   +    isPermission,
+        PolicyDecision,
+     } from '@backstage/plugin-permission-common';
+   ```
+
+5. Then you can add the following:
+
+```ts
+class ExamplePermissionPolicy implements PermissionPolicy {
```

Comment:
I think I'll try and refactor this so that it also allows the owner of the entity to do the refresh but I think I need to change how this permission is setup?

Reply:
Seeing as I've taken this out we won't need to worry about this for now